esphome/components/light/base_light_effects.h

```diff
@@ -11,6 +11,40 @@ inline static float random_cubic_float() {
   return r * r * r;
 }
 
+/// Pulse effect.
+class PulseLightEffect : public LightEffect {
+ public:
+  explicit PulseLightEffect(const std::string &name) : LightEffect(name) {}
+
+  void apply() override {
+    const uint32_t now = millis();
+    if (now - this->last_color_change_ < this->transition_length_) {
```

Comment:
I think this is supposed to be `update_interval_`
```suggestion
    if (now - this->last_color_change_ < this->update_interval_) {
```

Reply:
Fixed.